n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Following code is the implementation of the N-queen problem.
+ * The solution is to place the queens on board such that no queen is under attack by another queen.
+ * There can be a number of possible solutions for a given length of board.
+ * This implementation prints only one valid solution, it can be extended to print all possible valid solutions.
+ */
+
+import java.util.Arrays;
+
+public class NQueenProblem {
+
+    public static boolean isQueenSafe(int[][] board, int row, int column) {
+        int i;
+        int j;
+        int boardLength = board[0].length;
+        for (i = 0; i < column; ++i)
+            if (board[row][i] == 1)
+                return false;  // If there's another Queen present in the same row.
+
+        for (i = row, j = column; i >= 0 && j >= 0; --i, --j)
+            if (board[i][j] == 1)
+                return false;  // If there's another Queen present in the upper diagonal.
+
+        for (i = row, j = column; j >= 0 && i < boardLength; ++i, --j)
+            if (board[i][j]==1)
+                return false;  // If there's another Queen present in the lower diagonal.
+
+        return true;
+    }
+
+    public static boolean solveNQueen(int[][] board, int column) {
```

Comment:
This implementation needs not to be exposed. Because user would not be calling `solveNQueen(board, 0)`, as the second parameter is not relevant to him.

You may better introduce another function and make it `public` - 
```java
public static final boolean solveNQueen(int[][] board) {
        return solveNQueen(board, 0);
}
```